packages/mobile/src/harmony-native/foundations/theme/theme.ts

```diff
@@ -1,29 +1,235 @@
-import type { ColorTheme } from '../color'
-import { colorTheme } from '../color'
-import type { Theme } from '../types'
-import type { Typography } from '../typography'
-import { typography } from '../typography'
+import {
+  themes as harmonyThemes,
+  primitiveTheme as harmonyPimitiveTheme
+} from '@audius/harmony'
+import { mapValues } from 'lodash'
 
-export type HarmonyTheme = {
-  type: Theme
-  color: ColorTheme[keyof ColorTheme]
-  typography: Typography
+// linear-gradient at 315deg
+const baseLinearGradient = {
+  start: { x: 0, y: 1 },
+  end: { x: 1, y: 0 }
 }
 
-export const dayTheme: HarmonyTheme = {
-  type: 'day',
-  color: colorTheme.day,
-  typography
+const primitiveOverrides = {
+  day: {
+    special: {
+      gradient: {
+        ...baseLinearGradient,
+        colors: ['#5B23E1', '#A22FEB']
+      }
+    }
+  },
+  dark: {
+    special: {
+      gradient: {
+        ...baseLinearGradient,
+        colors: ['#7652CC', '#B05CE6']
+      }
+    }
+  },
+  matrix: {
+    special: {
+      gradient: {
+        ...baseLinearGradient,
+        colors: ['#7652CC', '#B05CE6']
+      }
+    }
+  }
 }
 
-export const darkTheme: HarmonyTheme = {
-  type: 'dark',
-  color: colorTheme.dark,
-  typography
+export const primitiveTheme = {
+  day: {
+    ...harmonyPimitiveTheme.day,
+    special: {
+      ...harmonyPimitiveTheme.day.special,
+      ...primitiveOverrides.day.special
+    }
+  },
+  dark: {
+    ...harmonyPimitiveTheme.dark,
+    special: {
+      ...harmonyPimitiveTheme.dark.special,
+      ...primitiveOverrides.dark.special
+    }
+  },
+  matrix: {
+    ...harmonyPimitiveTheme.matrix,
+    special: {
+      ...harmonyPimitiveTheme.matrix.special,
+      ...primitiveOverrides.matrix.special
+    }
+  }
 }
```

Comment:
so down, except im worried about the types not being exactly right, since merge can munge deep merging on type basis. wil try this out and report back though, cause agreed the above wouldnt be the approach i would normally take

Reply:
ended up working great for that part, good call :)